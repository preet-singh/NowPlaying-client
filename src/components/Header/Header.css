@media screen and (min-width:320px){

    .Header {
        color: white;
    }
}