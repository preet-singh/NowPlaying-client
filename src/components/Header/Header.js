//Dependencies
import React from 'react';
import { Link } from 'react-router-dom';

//Components
import LoginButton from '../../components/LoginButton/LoginButton';

//CSS
import './Header.css';

//images
import glowingWhite from '../../assets/white-play-button-glow.png'

function Header(props) {
  return(
    <div className="Header">
      <Link to='/'><h1><span>N<img src={glowingWhite} className="header-o" alt="glowing o"/>W </span> PLAYING</h1></Link>
      <div className="under-bar"><Link to="" onClick={(e) => console.log('delete about local storage')} className="about-link">ABOUT</Link><LoginButton classNames="log-button" /></div>
    </div>

  );
}

export default Header;