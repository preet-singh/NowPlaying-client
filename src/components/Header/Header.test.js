import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Header testing', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <MemoryRouter>
            <Header />
        </MemoryRouter>, 
    div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the header', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
});