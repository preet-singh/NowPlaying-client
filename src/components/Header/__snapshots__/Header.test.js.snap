// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testing renders the header 1`] = `
<div
  className="Header"
>
  <Link
    to="/"
  >
    <h1>
      <span>
        N
        <img
          className="header-o"
          src="https://i.imgur.com/uTCb1eb.png"
        />
        W 
      </span>
       PLAYING
    </h1>
  </Link>
  <LoginButton
    classNames="header-login mobile-hide"
  />
  <LoginButton
    classNames="mobile-login tablet-hide"
  />
</div>
`;
